components/cdc/src/endpoint.rs

```diff
@@ -686,14 +686,8 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
             let resps = futures03::future::join_all(regions).await;
             let regions = resps
                 .into_iter()
-                .filter(|resp| match resp {
-                    Either::Left(()) => false,
-                    Either::Right(_) => true,
-                })
-                .map(|resp| match resp {
-                    Either::Left(()) => unreachable!(),
-                    Either::Right(region_id) => region_id,
-                })
+                .filter(|resp| resp.is_some())
+                .map(|resp| resp.unwrap())
```

Comment:
What about `filter_map`?
```suggestion
                .filter_map(|resp| resp)
```